app/main.py

```
@@ -1,7 +1,36 @@
-def get_weather() -> None:
-    # write your code here
-    pass
+import os
 
+import requests
+from fastapi import FastAPI, HTTPException
 
-if __name__ == "__main__":
-    get_weather()
+app = FastAPI()
+
+URL = "http://api.weatherapi.com/v1/current.json?"
+API_KEY = os.environ.get("API_KEY")
+CITY = "Paris"
+
+
+@app.get("/weather")
+def get_weather() -> any:
+    if not API_KEY:  # Check if API_KEY is missing
+        raise HTTPException(
+            status_code=500,
+            detail="API_KEY is missing. Set the environment variable."
+        )
+    response = requests.get(URL + f"key={API_KEY}&q={CITY}")
```

You should wrap the `requests.get` call in a try-except block to handle network-related exceptions (such as connection errors or timeouts) from the external API. This is required by the checklist: 'Wrap the requests.get call in a try-except block to gracefully handle network-related exceptions and return a clear error message to the user.'

Currently, if a network error occurs, the application will raise an unhandled exception instead of returning a user-friendly error message.